File: lv2/JadenCase 문자열 만들기/code.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string s) {
    string answer = "";
    // 예외처리1 : 첫글자 소문자 처리
    answer += ('a' <= s[0] && s[0] <= 'z') ? s[0] -'a' + 'A' : s[0];

    for(int i = 1; i < s.size(); i++){
        // 예외처리2 : 앞 글자 공백, 뒷 글자 소문자 
        if(s[i-1] == ' ' && ('a' <= s[i] && s[i] <= 'z')){
            answer += s[i] -'a' + 'A';
        // 예외처리3 : 앞 글자 알파벳, 뒷 글자 대문자
        } else if(s[i-1] != ' ' && ('A' <= s[i] && s[i] <= 'Z')){
            answer += s[i] -'A' + 'a';
        } else
            answer += s[i];
    }
    return answer;
}

File: lv1/가장 가까운 같은 글자/code.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<Character, Integer> map = new HashMap<>();
        
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            Integer prev = map.put(c,i);
            if(prev == null){
                answer[i] = -1;
            } else {
                answer[i] = i - prev;
            }
        }
        return answer;
    }
}
